tapeagents/llms.py

```diff
@@ -578,18 +612,17 @@ def _generate(self, prompt: Prompt) -> Generator[LLMEvent, None, None]:
             data = r.json()
             try:
                 content = data["choices"][0]["message"]["content"]
-                if self.remove_leading_white_space:
-                    # vllm sometimes adds a whitespace at the beginning of the completion
-                    if content[0] == " ":
-                        content = content[1:]
-                    else:
-                        logger.error(f"Expected leading white space in completion: \n{content}")
                 if not content:
                     logger.warning(f"Empty completion {data}")
 
                 logprobs = None
                 if self.collect_logprobs:
-                    logprobs = data["choices"][0]["logprobs"]["content"]
+                    prompt_token_ids = self.tokenizer.apply_chat_template(
+                        prompt.messages, add_special_tokens=True, add_generation_prompt=True
+                    )
+                    # prompt_decoded = self.tokenizer.decode(prompt_token_ids, skip_special_tokens=False)
+                    completion_logprobs = data["choices"][0]["logprobs"]["content"]
+                    logprobs = self.process_logprobs(prompt_token_ids, completion_logprobs)
```

Comment:
There's something that makes me nervous in this code: we remove eos from `content`, but not from the list of tokens  in `logprobs`. That's a bit bizarre.